ssl/ech/ech_local.h

```diff
@@ -207,6 +225,39 @@ typedef struct ossl_ech_conn_st {
     unsigned char client_random[SSL3_RANDOM_SIZE]; /* CH random */
 } OSSL_ECH_CONN;
 
+/* Return values from ech_same_ext */
+#  define OSSL_ECH_SAME_EXT_ERR 0 /* bummer something wrong */
+#  define OSSL_ECH_SAME_EXT_DONE 1 /* proceed with same value in inner/outer */
+#  define OSSL_ECH_SAME_EXT_CONTINUE 2 /* generate a new value for outer CH */
+
+/*
+ * During extension construction (in extensions_clnt.c and surprisingly also in
+ * extensions.c), we need to handle inner/outer CH cloning - ech_same_ext will
+ * (depending on compile time handling options) copy the value from CH.inner to
+ * CH.outer or else processing will continue, for a 2nd call, likely generating
+ * a fresh value for the outer CH. The fresh value could well be the same as in
+ * the inner.
+ *
+ * This macro should be called in each _ctos_ function that doesn't explicitly
+ * have special ECH handling.
+ *
+ * Note that the placement of this macro needs a bit of thought - it has to go
+ * after declarations (to keep the ansi-c compile happy) and also after any
+ * checks that result in the extension not being sent but before any relevant
+ * state changes that would affect a possible 2nd call to the constructor.
+ * Luckily, that's usually not to hard, but it's not mechanical.
+ */
+#  define ECH_IOSAME(s, pkt) \
+    if (s->ext.ech.es != NULL && s->ext.ech.grease == 0) { \
+        int __rv = ech_same_ext(s, pkt); \
```

Comment:
Never use double underscore at the beginning of symbols - these are reserved to the compiler and libc.

Reply:
ack